### ISSUE TITLE
Bad icon size system tray KDE

### ISSUE THREAD

**Issue by gusarg81** [yakyak/yakyak#788]
Hi,

    There is a problem with the system tray icon size, which looks huge. This is on KDE desktop (plasma desktop).

KDE Frameworks: 5.39.0  
KDE Plasma: 5.11.1
Distro: KDE Neon (Ubuntu 16.04 based)

![yakyak_systray_icon_kde](https://user-images.githubusercontent.com/6312430/31961783-269305f4-b8d2-11e7-8dd9-a3e7db1ca207.png)

**Comment by copyme** [comment]
Hmm, I do not have this problem on openSUSE Leap 42.3.

![spectacle ns6782](https://user-images.githubusercontent.com/456607/32135345-e3c86dac-bbfd-11e7-930f-09b717d03194.png)

**Comment by gusarg81** [comment]
So, this app only works "right" with OpenSuse or specific version of KDE? (since you are not mentioning the KDE version used in OpenSuse, which is a distro I don't use).

Thanks.

**Comment by copyme** [comment]
@gusarg81 Sorry, I should have added such important information. So, openSUSE 42.3 has KDE5 LTS 5.8.7.

**Comment by copyme** [comment]
@gusarg81 maybe the issues is related somehow to the Plasma Workspace theme which you use.

**Comment by gusarg81** [comment]
#copyme well, I don't think so, since is the default plasma theme (Breeze).

**Comment by gusarg81** [comment]
BTW, is you want to try at least in a Virtualbox, you can get the iso image in https://neon.kde.org/download. It is the most official KDE Distro/DE from KDE itself.

**Comment by gusarg81** [comment]
Hi,

    Any news about this? 

Thanks.

**Comment by averissimo** [comment]
Some KDE plasma *(versions/themes/something else)* seems not to scale the icons in tray -_- gnome and unity do it without a problem

I'm trying the virtualbox image to see if I can reproduce it. 

If so we'll have to either detect kde and change icon size, or wait for some solution from electron upstream

**Comment by averissimo** [comment]
this bug report from electron could be relevant: https://github.com/electron/electron/issues/10163

**Comment by gusarg81** [comment]
Well, as I said I am using thw default theme and is the only app that has problems like this (at least all apps that I use which has systray icon)

**Comment by averissimo** [comment]
Seems like electron is using an old api, which could just force the size of the original icon instead of scaling.

Just for the record, doing `process.env.DESKTOP_SESSION == 'kde'` could be used to force a specific size for tray icon.. **although this is an ugly solution which I would prefer to avoid.**

@gusarg81 or someone else, can you find out what is the default size for icons in tray in kde?

**Comment by gusarg81** [comment]
Sure, I can digg on it but have in mind that KDE panel is resizable.

**Comment by averissimo** [comment]
Nevermind then.

I wish svg could be used, but last time I checked it couldn't. All our icons are in svg format so that would solve it. I'll give it a go just in case something has changed :)

**Comment by sosguthorpe** [comment]
The issues with KDE icons not resizing and also not allowing the application to restore on click seems to be down to the [libappindicator library](https://launchpad.net/libappindicator). This library, for me was only installed as a result of installing the Google Chrome browser. It is supposed to play nicely in KDE, but either the electron usage of this library or the library itself isn't scaling properly.

If the library is removed then the default system tray icons in KDE are used and the Icon both looks correct and allows for the left-click event to be caught and therefore allows restore of the application using left-click. Obviously, removal isn't an option for me, and probably others noticing this behaviour, because it's usually only installed, in KDE environments, as a dependency of some other application.

TLDR: The best workaround IMHO is to install a NOOP package for the libappindicator library under KDE, as the result will be that the default KDE implementation will be used instead, and the library remains present to satisfy the dependencies. [This comment](https://github.com/vector-im/riot-web/issues/3133#issuecomment-314980590) on a related issue provides a pre-built deb version for easy install on Debian based systems.

I have, thus far, not noticed any undesirable side-effects as of yet.

**Comment by dos1** [comment]
It's actually Chromium that's breaking the icon. I've submited a patchset that fixes this issue: https://chromium-review.googlesource.com/c/chromium/src/+/1173235

**Comment by akontsevich** [comment]
Hmm...  @sosguthorpe it works for libappindicator v1 and fails for v3. But version 3 required for NM-applet, pasystray, mate DE, etc. So think noop package will not work, or not exists for many distributions: I have openSUSE for example.

**Comment by dos1** [comment]
Having libappindicator installed should be fine, as long as it doesn't expose the "Activate" method on its dbus interface (some patched versions, like the one in Arch, did, which was breaking it on recent Plasmas).
